uswds/uswds
# USWDS: style aria-disabled to match disabled

## Description

Closes: https://github.com/uswds/uswds/issues/4508

Preview links:
- [Button](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-aria-disabled/?path=/story/components-button--default)
- [Button Group](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-aria-disabled/?path=/story/components-button-group--default)
- [Checkbox](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-aria-disabled/?path=/story/components-form-inputs-checkbox--default)
- [Date Picker](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-aria-disabled/?path=/story/components-form-inputs-date-picker--aria-disabled)
- [Radio](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-aria-disabled/?path=/story/components-form-inputs-radio--default)

### Problem
At times, users of the Design System will want to use `aria-disabled` in place of the `disabled` attribute, but are limited because currently there is no explicit styling for `aria-disabled` elements in the Design System. 

`aria-disabled` provides a different experience to end-users that use keyboard navigation or assistive technologies. The primary difference between the two attributes is this:
-  `disabled` elements _do not_ receive focus on tab, which can result in the item being:
    - invisible to users on screen readers who use tab to navigate forms
    - difficult and confusing to interact with for other users using keyboard navigation without a screen reader, particularly for those with cognitive and fine-motor disabilities 
- `aria-disabled` _does_ receive focus on tab, which can give users important clues about their next steps

The movement in industry conversation seems to be towards `aria-disabled` as the preferred user experience  (See resources below), but other mainstream design systems still use the `disabled` attribute. This item is only one piece of a larger discussion about the UX/accessibility of disabled buttons. 

From [Mozilla](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled):
> The aria-disabled attribute is similar to the HTML disabled attribute, but better. The first rule of ARIA is "If you can use a native HTML element or attribute with the semantics and behavior you require already built in, instead of repurposing an element and adding an ARIA role, state or property to make it accessible, then do so." aria-disabled is an exception to that rule. It is one of the only cases where an ARIA attribute may be better than the native disabled attribute HTML equivalent.

#### Resources
- [Making disabled buttons more inclusive](https://css-tricks.com/making-disabled-buttons-more-inclusive/) (CSS Tricks)
- [aria-disabled](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled) (Mozilla)
- [Videos comparing experience of disabled and aria-disabled](https://a11y-101.com/development/aria-disabled) (A11y-101)
### Solution
Providing styling for both `aria-disabled` and `disabled` will give users more flexibility to make the appropriate usability decisions for their site. 

:warning: This PR is concerned only with CSS styling of `aria-disabled`. It is not intended to replace `disabled` as the default attribute in our components.

### Tasks completed
- [x] Add styling for aria-disabled wherever `:disabled` is declared
- [x] Standardize hover states for disabled/aria-disabled buttons 
    - Note: This likely addresses the concerns in: https://github.com/uswds/uswds/issues/4420
- [x] Set disabled/aria-disabled cursor to `auto` instead of `pointer`
    - Note: Radio and Checkbox do use `cursor: not-allowed`. Should we standardize to one value?
- [x] Add aria-disabled control to relevant stories to help with testing

### Testing
1. Confirm that all `:disabled` SCSS references also include `aria-disabled`
1. Check that the styling for `disabled` and `aria-disabled` align on the following components:
    - Button
    - Button group
    - Date picker 
    - Radio
    - Checkbox
    
## Additional information

Before you hit Submit, make sure you’ve done whichever of these applies to you:
